datafusion/expr/src/expr.rs

```diff
@@ -1212,11 +1213,22 @@ impl Expr {
     /// let expr = col("foo").alias("bar").alias("baz");
     /// assert_eq!(expr.unalias_nested(), col("foo"));
     /// ```
-    pub fn unalias_nested(self) -> Expr {
-        match self {
-            Expr::Alias(alias) => alias.expr.unalias_nested(),
-            _ => self,
-        }
+    pub fn unalias_nested(self) -> Transformed<Expr> {
```

Comment:
this is an API change (it returns Transformed rather than just Expr) but I think that makes more sense